sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -300,6 +300,11 @@ def build_task_spec_for_task(
                 'str, int, float, bool, dict, and list.'
                 f'Got {input_value} of type {type(input_value)}.')
 
+        if task.ignore_upstream_failure_tag:
```

Comment:
I think this is over-indented one level and is called once for each input.

Reply:
Good catch! Thanks